script.js

```diff
@@ -1,4 +1,208 @@
+//define game mode
+var game_state_select_num_of_players = "select number of players";
+var game_state_waiting_for_username = " waiting for player username";
+var game_state_select_num_of_dice = "player select num of dice";
+var game_state_roll_dice = "player roll your dice";
+var game_state_autogenerate_order = "player computer autogenerates order";
+var game_state_find_winner = "find winner";
+var currentGameState = game_state_select_num_of_players;
+
+var currentPlayer = 0;
+var game_mode_setting = "select normal or reverse";
+var game_mode_normal = "highest";
+var game_mode_reverse = "lowest";
+var currentGameMode = game_mode_setting;
+
+//define global variables
+var diceLists = [];
+var diceSelected = "";
+var userName = "";
+var totalPlayers = 0;
+var playerList = [];
+
+var numOfDiceSelected = function (num) {
+  var numOfDice = num;
+  if (num != Number(num)) {
+    return -1;
+  } else {
+    return Number(numOfDice);
```

Comment:
an alternate way you can write this is 
if (num!=Number(num)){
   return -1
}
return Number(numOfDice)

This way you can eliminate the else statement. This is possible as the if statement already contains a return hence only in scenarios where the if statement condition is not fulfilled will the function return Number(numOfDice). In cases where the if condition is met, JS will exit the function after returning -1 and would not run the second return statement